src/NHibernate/Hql/Ast/HqlTreeBuilder.cs

```diff
@@ -416,7 +416,7 @@ public HqlFalse False()
 			return new HqlFalse(_factory);
 		}
 
-		public HqlIn In(HqlExpression itemExpression, HqlTreeNode source)
+		public HqlIn In(HqlExpression itemExpression, params HqlTreeNode[] source)
```

Comment:
I've found the proper solution for not changing this. You need to use `HqlTreeBuilder.ExpressionSubTreeHolder`.